restdocs-api-spec-jsonschema/src/test/kotlin/com/epages/restdocs/apispec/jsonschema/JsonSchemaFromFieldDescriptorsGeneratorTest.kt

```diff
@@ -230,6 +250,45 @@ class JsonSchemaFromFieldDescriptorsGeneratorTest {
         thenSchemaValidatesJson("""{ some: "ENUM_VALUE_1" }""")
     }
 
+    @Test
+    fun should_generate_schema_for_top_level_array_with_size_constraint() {
+        givenFieldDescriptorWithTopLevelArrayOfAnyWithSizeConstraint()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).minItems).isEqualTo(1)
+        then((schema as ArraySchema).maxItems).isEqualTo(255)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_generate_schema_for_top_level_array_of_any_with_size_constraint() {
+        givenFieldDescriptorWithTopLevelArrayOfArrayOfAnyWithSizeConstraint()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(ArraySchema::class.java)
+        then(((schema as ArraySchema).allItemSchema as ArraySchema).minItems).isEqualTo(1)
+        then(((schema as ArraySchema).allItemSchema as ArraySchema).maxItems).isEqualTo(255)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_generate_schema_for_unspecified_array_contents_with_size_constraint() {
```

Comment:
`should_generate_schema_for_unspecified_array_contents_with_size_constraint`
->
`should_generate_schema_for_array_with_size_constraint`

and I suggest to move this test below the first new test you introduced (`should_generate_schema_for_required_object_in_array`), since both target an array field in a top-level object. Once as a required field, once with size constraints.